src/System Application/App/Agent/Setup/AccessControls/SelectAgentPermissionsPart.Page.al

```diff
@@ -0,0 +1,220 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Agents;
+
+using System.Security.AccessControl;
+
+page 4340 "Select Agent Permissions Part"
+{
+    PageType = ListPart;
+    ApplicationArea = All;
+    SourceTable = "Access Control Buffer";
+    SourceTableTemporary = true;
+    Caption = 'Agent permissions';
+    MultipleNewLines = false;
+    Editable = true;
+    DataCaptionExpression = '';
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    layout
+    {
+        area(Content)
+        {
+            repeater(Main)
+            {
+                Caption = 'Agent permissions';
+
+                field(PermissionSet; Rec."Role ID")
+                {
+                    ApplicationArea = Basic, Suite;
+                    Caption = 'Permission Set';
+                    ToolTip = 'Specifies the ID of a security role that has been assigned to this Windows login in the current database.';
+                    Style = Unfavorable;
+                    StyleExpr = PermissionSetNotFound;
+
+                    trigger OnLookup(var Text: Text): Boolean
+                    var
+                        LookupPermissionSet: Page "Lookup Permission Set";
+                    begin
+                        LookupPermissionSet.LookupMode := true;
+                        if LookupPermissionSet.RunModal() = ACTION::LookupOK then begin
+                            LookupPermissionSet.GetRecord(PermissionSetLookupRecord);
+                            Text := PermissionSetLookupRecord."Role ID";
+                            PermissionSetLookupRecord.SetRecFilter();
+                            exit(true);
+                        end;
+                    end;
+
+                    trigger OnValidate()
+                    begin
+                        PermissionSetLookupRecord.SetRange("Role ID", Rec."Role ID");
+                        if PermissionSetLookupRecord.FindFirst() then begin
+                            if PermissionSetLookupRecord.Count() > 1 then
+                                Error(MultipleRoleIDErr, Rec."Role ID");
```

Comment:
I'm not sure what will happen to global variables, error will revert all of the database changes.